### DISCUSSION TITLE
airflow/configuration.py [logging] log_filename_template FutureWarning

### DISCUSSION THREAD

**Discussion by kolfild26** [apache/airflow#23646] [Q&A] [answered]
### Apache Airflow version

2.3.0 (latest released)

### What happened

Hi,
Tried to upgrade from v.2.2.3 to v.2.3.

`airflow db upgrade:`
```

./airflow2.3/lib/python3.8/site-packages/airflow/configuration.py:412:
FutureWarning: The 'log_filename_template' setting in [logging] has the old default value of '{{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log'. This value has been changed to 'dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log' in the running config, but please update your config before Apache Airflow 3.0.
```

I reviewed my airflow.cfg file and changed what was recommended by the waring message. And also checked [the default config file](https://github.com/apache/airflow/blob/main/airflow/config_templates/default_airflow.cfg) in the airflow github repository. Here is the same string:
`log_filename_template = dag_id={{{{ ti.dag_id }}}}/run_id={{{{ ti.run_id }}}}/task_id={{{{ ti.task_id }}}}/{{%% if ti.map_index >= 0 %%}}map_index={{{{ ti.map_index }}}}/{{%% endif %%}}attempt={{{{ try_number }}}}.log`

But after I run again I get the exception:
`airflow db upgrade --show-sql-only:`
```
Unable to load the config, contains a configuration error.
Traceback (most recent call last):
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/logging/config.py", line 563, in configure
    handler = self.configure_handler(handlers[name])
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/logging/config.py", line 744, in configure_handler
    result = factory(**kwargs)
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/airflow/utils/log/file_task_handler.py", line 51, in __init__
    self.filename_template, self.filename_jinja_template = parse_template_string(filename_template)
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/airflow/utils/helpers.py", line 176, in parse_template_string
    return None, jinja2.Template(template_string)
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/jinja2/environment.py", line 1195, in __new__
    return env.from_string(source, template_class=cls)
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/jinja2/environment.py", line 1092, in from_string
    return cls.from_code(self, self.compile(source), gs, None)
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/jinja2/environment.py", line 757, in compile
    self.handle_exception(source=source_hint)
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<unknown>", line 1, in template
jinja2.exceptions.TemplateSyntaxError: expected token ':', got '}'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/media/data/anaconda3/envs/airflow2.3/bin/airflow", line 5, in <module>
    from airflow.__main__ import main
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/airflow/__init__.py", line 47, in <module>
    settings.initialize()
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/airflow/settings.py", line 561, in initialize
    LOGGING_CLASS_PATH = configure_logging()
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/airflow/logging_config.py", line 73, in configure_logging
    raise e
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/site-packages/airflow/logging_config.py", line 68, in configure_logging
    dictConfig(logging_config)
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/logging/config.py", line 808, in dictConfig
    dictConfigClass(config).configure()
  File "/media/data/anaconda3/envs/airflow2.3/lib/python3.8/logging/config.py", line 570, in configure
    raise ValueError('Unable to configure handler '
ValueError: Unable to configure handler 'task'
```
it seems that it is the **log_filename_template** parameter causes the error as:

- there are no any other exceptions or error messages
- exception disapears when reverting the changes

I don't really know whether this FutureWarning causes any problems with working with airflow. And is it okay to stay with the old value of the **log_filename_template**?

### How to reproduce

1. upgrade airflow installation from [pip apache airflow](https://pypi.org/project/apache-airflow/)
2. run `airflow db upgrade`

### Operating System

Oracle Linux 7.8

### Deployment

Virtualenv installation

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by kolfild26** [comment]
`log_filename_template = dag_id=...`
This looks strange, btw. I didn't find any other lines containing two `=` in airflow.cfg.

Then tried to delete `"dag_id="` but with no success. The exception remains.

**Comment by uranusjr** [comment, marked as answer]
I don’t think you should double the brackets and stuff. Use normal Jinja2 syntax instead.

**Reply by maslick** [reply]
This should be the correct answer:
change this (old format Airflow v2):
```
log_filename_template = {{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log
```

into this (new format Airflow v3):
```
log_filename_template = dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{%% if ti.map_index >= 0 %%}map_index={{ ti.map_index }}/{%% endif %%}attempt={{ try_number }}.log
```

**Reply by Taragolis** [reply]
> into this (Airflow v3)

🫨

**Reply by maslick** [reply]
@Taragolis have I made a mistake? I just see this post is marked as answer, but doesn't contain the answer. So I explicitly noted it down for anyone who come across this issue (warning message).

<img width="613" alt="Screenshot 2023-09-13 at 18 14 23" src="https://github.com/apache/airflow/assets/5644706/68ccecf9-be33-4e5a-98bb-e20a4e7c2004">

When running Airflow 2.6.3 I get this warning message:
```
/usr/local/airflow/.local/lib/python3.10/site-packages/airflow/configuration.py:479 FutureWarning: The 'log_filename_template' setting in [logging] has the old default value of '{{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log'. This value has been changed to 'dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log' in the running config, but please update your config before Apache Airflow 3.0.
```

**Reply by Taragolis** [reply]
Just Airflow 3 not planned yet.
I guess youre sample about pre-2.3 and 2.3+ or whatever version default logging format changed

**Reply by maslick** [reply]
@Taragolis maybe Airflow 3 is not planned yet, but there's a warning message already in Airflow 2.6.3 (see my previous post)

**Reply by Taragolis** [reply]
It just say that old one would not work in 3.0. But current one introduced and preferable since airflow 2.3 https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-3-0-2022-04-30

And current one could be changed before Airflow 3 released

**Reply by maslick** [reply]
I understand what it says, I just don't want this warning message pop up in the logs. Thought it would be of help to someone :)

**Comment by kolfild26** [comment]
@uranusjr thanks
`log_filename_template = dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{%% if ti.map_index >= 0 %%}map_index={{ ti.map_index }}/{%% endif %%}attempt={{ try_number }}.log`
 <- this seems to be working.
The only thing, [the default source config](https://github.com/apache/airflow/blob/main/airflow/config_templates/default_airflow.cfg#L353) has wrong syntax

**Reply by uranusjr** [reply]
The default value there uses four brackets because it is supposed to be rendered, I believe, not to be read directly by a human.

**Reply by ginolegigot** [reply]
Yes i found the same fix.
https://github.com/apache/airflow/blob/main/airflow/config_templates/default_airflow.cfg this file I took as model is therefore wrong too

**Reply by kolfild26** [reply]
@ginolegigot 
`this file I took as model is therefore wrong too`
yes, that's what i was trying to explain. It's supposed to be rendered, but actually not being rendered.

**Reply by ginolegigot** [reply]
yes, thanks

**Comment by kolfild26** [comment]
@uranusjr 
one more. 
As I posted before, /airflow/configuration.py also has wrong suggested value in its warning message
```
./airflow2.3/lib/python3.8/site-packages/airflow/configuration.py:412:
FutureWarning: The 'log_filename_template' setting in [logging] has the old default value of '{{ ti.dag_id }}/{{ ti.task_id }}/{{ ts }}/{{ try_number }}.log'. This value has been changed to 'dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log' in the running config, but please update your config before Apache Airflow 3.0.
```
They differ, but both are not working.

**Comment by boykeu** [comment]
after un-double the bracket, airflow db init working, but when i tried to run the dag, then reproduce this error

{{standard_task_runner.py:92}} ERROR - Failed to execute job 931 for task test_ssh_operator (invalid interpolation syntax in '{{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{% if ti.map_index >= 0 %}map_index={{ ti.map_index }}/{% endif %}attempt={{ try_number }}.log' at position 65; 139422)

**Reply by ginolegigot** [reply]
Try this instead:
```jinja
log_filename_template = dag_id={{ ti.dag_id }}/run_id={{ ti.run_id }}/task_id={{ ti.task_id }}/{%% if ti.map_index >= 0 %%}map_index={{ ti.map_index }}/{%% endif %%}attempt={{ try_number }}.log
```
